gnomad/utils/annotations.py

```diff
@@ -1138,3 +1198,219 @@ def hemi_expr(
+    if vrs_only:
+        return vrs_dict
+
+    # Create a list to then add the dictionaries for frequency reports for
+    # different ancestry groups to.
+    list_of_group_info_dicts = []
+
+    # Define function to return a frequency report dictionary for a given group
+    def _create_group_dicts(
+        variant_ht: hl.Table,
+        group_index: int,
+        group_id: str,
+        group_label: str,
+        vrs_id: str,
+    ) -> dict:
+        """
+        Return a dictionary for the frequency information of a given variant for a given subpopulation.
+
+        :param variant_ht: Hail Table with only one row, only containing the desired variant.
+        :param group_index: Index of frequency within the 'freq' annotation containing the desired group.
+        :param group_id: String containing variant, genetic ancestry group, and sex (if requested). Example: "chr19-41094895-C-T.afr.XX".
+        :param group_label: String containing the full name of genetic ancestry group requested. Example: "African/African American".
+        :param vrs_id: String containing the VRS ID of the variant in ht_subpop.
+        :return: Dictionary containing VRS information (and genetic ancestry group if desired) for specified variant.
+
+        """
+        # Obtain frequency information for the specified variant
+        group_freq = variant_ht.freq[group_index]
+
+        # Dictionary to be returned containing information for a specified group
+        freq_record = {
+            "id": group_id,
+            "type": "PopulationAlleleFrequency",
+            "label": f"{group_label} Population Allele Frequency for {variant}",
+            "focusAllele": vrs_id,
+            "focusAlleleCount": group_freq["AC"].collect()[0],
+            "locusAlleleCount": group_freq["AN"].collect()[0],
+            "alleleFrequency": group_freq["AF"].collect()[0],
+            "population": f"{label_name}{label_version}:{group_id}",
+            "ancillaryResults": {
+                "homozygotes": group_freq["homozygote_count"].collect()[0]
+            },
+        }
+
+        return freq_record
+
+    # Iterate through provided groups and generate dictionaries
+    if ancestry_groups:
+        for group in ancestry_groups:
+            key = f"{group}-adj"
+            index_value = ht.freq_index_dict.get(key)
+            group_result = _create_group_dicts(
+                variant_ht=ht,
+                group_index=index_value,
+                group_id=group,
+                group_label=ancestry_groups_dict[group],
+                vrs_id=vrs_id,
+            )
+
+            # If specified, stratify group information by sex.
+            if by_sex:
+                sex_list = []
+                for sex in ["XX", "XY"]:
+                    sex_key = f"{group}-{sex}-adj"
+                    sex_index_value = ht.freq_index_dict.get(sex_key)
+                    sex_label = f"{group}.{sex}"
```

Comment:
add variant to this label

Reply:
variant added to label in comment below, resolving